src/wasm32/ffi.c

```diff
@@ -239,9 +256,6 @@ ffi_call, (ffi_cif * cif, ffi_fp fn, void *rvalue, void **avalue),
       SIG(sig += 'i');
       break;
     case FFI_TYPE_UINT16:
-      args.push(DEREF_U16(arg_ptr, 0));
```

Comment:
Does it matter whether we use `DEREF_U16` or `DEREF_I16`? What about `DEREF_U32` vs `DEREF_I32` and `DEREF_U64` vs `DEREF_I64? Probably does the same thing either way? We're inconsistent about this.

Reply:
This was optimized since `DEREF_U16` and `DEREF_I16` were both defined as:
```c
#define DEREF_U16(addr, offset) HEAPU16[(addr >> 1) + offset]
#define DEREF_I16(addr, offset) HEAPU16[(addr >> 1) + offset]
``` 

Looking back, maybe I should have changed the definition of `DEREF_I16` instead (so that it operates on `HEAP16`).

Reply:
Yeah, this is wrong in any case:
```C
#define DEREF_I16(addr, offset) HEAPU16[(addr >> 1) + offset]
```